### ISSUE TITLE
future of scala ides

### ISSUE THREAD

**Issue by fommil** [sschaef/amora#28]
I'm not sure how to contact you, but I just wanted to say I'm sorry to see you leave Scala! And sorry to hear that ScalaIDE has not been funded.

I will not be far behind you, but I'll be staying in programming as it pays the bills very well and I still enjoy the technical challenges. Haskell for me, I think.

FYI with regards to scala ides, I think people did actually listen to what many of us had to say, although though they may not be explicitly saying so. There is now a working group, I would have nominated you if you were interested. I wrote up my opinions for them at http://ensime.github.io/lsp-wg/ but I am not a member as I didn't win a seat on the ballot.

TL;DR a build tool "metacp" can index all the classpath binaries into protobuf files meaning that IDEs do not need to read ASM or scalap, and be robust against scala versions. Scalameta can produce untyped syntactic trees for use for position refinement. In addition, a compiler plugin can produce type markup to turn those untyped trees into typed trees.

If I were writing ensime again, I would have done it this way. But I no longer have the energy. All current work seems to be happening at https://github.com/scalameta/metals

**Comment by kiritsuku** [comment]
Yeah, they will be doing fine without us ;)
